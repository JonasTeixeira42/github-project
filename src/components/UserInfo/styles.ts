import styled, { css } from 'styled-components'
import { darken } from 'polished'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    border-radius: ${theme.border.radius};
    box-shadow: ${theme.shadow.default};
    padding: ${theme.spacings.small};
    color: ${theme.colors.black};
  `}
`

export const Id = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
    color: ${theme.colors.gray};
  `}
`

export const Name = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    margin-bottom: ${theme.spacings.small};
  `}
`

export const Link = styled.a`
  ${({ theme }) => css`
    text-decoration: none;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.blue};
    padding-bottom: 0.2rem;
    border-bottom: 0.1rem solid ${theme.colors.blue};
    transition: all 0.3s;

    &:hover {
      color: ${darken(0.3, theme.colors.blue)};
      border-bottom: 0.1rem solid ${darken(0.3, theme.colors.blue)};
    }
  `}
`
