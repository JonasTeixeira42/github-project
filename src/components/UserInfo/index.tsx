import React from 'react'

import { Users } from 'hooks/use-users'

import * as S from './styles'

export type UserInfoProps = Omit<Users, 'avatar_url'>

const UserInfo = ({ id = 1, login = '', html_url = '' }: UserInfoProps) => {
  return (
    <S.Wrapper>
      <S.Id>{id}</S.Id>
      <S.Name>{login}</S.Name>
      <S.Link href={html_url} target="_blank">
        Go to profile
      </S.Link>
    </S.Wrapper>
  )
}

export default UserInfo
