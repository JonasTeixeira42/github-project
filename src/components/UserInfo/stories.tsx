import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import UserInfo, { UserInfoProps } from '.'

export default {
  title: 'UserInfo',
  component: UserInfo,
  args: {
    id: 1,
    login: 'UserTest',
    html_url: 'https://www.google.com/'
  }
} as Meta

export const Default: Story<UserInfoProps> = (args) => <UserInfo {...args} />
