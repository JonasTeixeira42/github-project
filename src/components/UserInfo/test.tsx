import React from 'react'
import { screen } from '@testing-library/react'

import UserInfo from '.'
import { renderWithTheme } from 'utils/tests/helpers'

const props = {
  id: 1,
  login: 'UserTest',
  html_url: 'https://www.google.com/'
}

describe('<UserInfo />', () => {
  it('should render the component correctly', () => {
    const { container } = renderWithTheme(<UserInfo {...props} />)

    expect(screen.getByRole('heading', { name: /1/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /UserTest/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /go to profile/i })
    ).toHaveAttribute('href', 'https://www.google.com/')

    expect(container.firstChild).toMatchSnapshot()
  })
})
