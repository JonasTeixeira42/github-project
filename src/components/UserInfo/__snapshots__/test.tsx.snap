// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> should render the component correctly 1`] = `
.c0 {
  background-color: #ffffff;
  border-radius: 0.4rem;
  box-shadow: 0px 3px 6px #00000029;
  padding: 2.4rem;
  color: #000000;
}

.c1 {
  font-size: 1.6rem;
  margin-bottom: 2.4rem;
  color: #808080;
}

.c2 {
  font-size: 1.6rem;
  margin-bottom: 2.4rem;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  color: #3799ff;
  padding-bottom: 0.2rem;
  border-bottom: 0.1rem solid #3799ff;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c3:hover {
  color: #004d9d;
  border-bottom: 0.1rem solid #004d9d;
}

<div
  class="c0"
>
  <h3
    class="c1"
  >
    1
  </h3>
  <h2
    class="c2"
  >
    UserTest
  </h2>
  <a
    class="c3"
    href="https://www.google.com/"
    target="_blank"
  >
    Go to profile
  </a>
</div>
`;
